engines/dfc_provider/app/services/supplied_product_builder.rb

```diff
@@ -13,16 +13,25 @@ def self.supplied_product(variant)
       description: variant.description,
       productType: product_type(variant),
       quantity: QuantitativeValueBuilder.quantity(variant),
+      spree_product_uri: id,
       spree_product_id: variant.product.id,
       image_url: variant.product&.image&.url(:product)
     )
   end
 
-  def self.import_variant(supplied_product)
+  def self.import_variant(supplied_product, host: "")
     product_id = supplied_product.spree_product_id
 
-    if product_id.present?
-      product = Spree::Product.find(product_id)
+    uri = RDF::URI.new(supplied_product.spree_product_uri)
+
+    if product_id.present? || uri.host == host
```

Comment:
I think I would check for identity of URL to be sure in this case. I found some nice helpers:
```rb
Rails.application.routes.recognize_path "/api/dfc/enterprises/1/supplied_products/4"
=> {:controller=>"dfc_provider/supplied_products", :action=>"show", :enterprise_id=>"1", :id=>"4"}

DfcProvider::Engine.routes.url_helpers.enterprise_supplied_product_url(
  Rails.application.routes.recognize_path "/api/dfc/enterprises/1/supplied_products/4"
)
=> "http://0.0.0.0:3000/api/dfc/enterprises/1/supplied_products/4"
```